src/NpcAdventureModApi.cs

```diff
@@ -0,0 +1,72 @@
+﻿using NpcAdventure.StateMachine;
+using NpcAdventure.StateMachine.State;
+using StardewValley;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace NpcAdventure
+{
+    public class NpcAdventureModApi
+    {
+        private NpcAdventureMod npcAdventureMod;
+
+        internal NpcAdventureModApi(NpcAdventureMod npcAdventureMod)
+        {
+            this.npcAdventureMod = npcAdventureMod;
+        }
+        public bool CanRecruit()
+        {
+            return npcAdventureMod != null && npcAdventureMod.CompanionManager != null && npcAdventureMod.CompanionManager.CanRecruit();
+        }
+        public string[] GetPossibleCompanions()
+        {
+            return npcAdventureMod.CompanionManager.PossibleCompanions.Keys.ToArray();
+        }
+        public bool IsPossibleCompanion(string npc)
+        {
+            return npcAdventureMod != null && npcAdventureMod.CompanionManager != null && npcAdventureMod.CompanionManager.PossibleCompanions != null && npcAdventureMod.CompanionManager.PossibleCompanions.ContainsKey(npc);
+        }
+        public bool CanRecruit(NPC npc)
+        {
+            return npcAdventureMod != null && npcAdventureMod.CompanionManager != null && npcAdventureMod.CompanionManager.PossibleCompanions != null && npcAdventureMod.CompanionManager.PossibleCompanions.TryGetValue(npc.Name, out CompanionStateMachine csm) && csm != null && npc != null && csm.Name == npc.Name && csm.CurrentStateFlag == CompanionStateMachine.StateFlag.AVAILABLE && csm.CanPerformAction();
```

Comment:
Also you can use question mark `?` for nullchecks here.

Reply:
sorry for all the bother, but if I use null checks here, I can't use csm, it won't compile

Reply:
```suggestion
            return this.npcAdventureMod?.CompanionManager?.PossibleCompanions?.TryGetValue(npc.Name, out CompanionStateMachine csm) && csm != null && npc != null && csm.Name == npc.Name && csm.CurrentStateFlag == CompanionStateMachine.StateFlag.AVAILABLE && csm.CanPerformAction();
```

It must work. Btw: use explicit `this` on instance class members. It's more readable and it's visible at first look this is an member field and not local variable.

Reply:
PS: You don't need nullcheck for everything here. CSM has always instantiated the filed Name or some other. If something it's null, then it'S probably something really bad wrong internally in NA.